File: backend/internal/model/terminal.go
package model

import "time"

type Terminal struct {
	ID                  int64     `json:"id"`
	SerialNumber        string    `json:"serial_number" binding:"required"`
	InstallationAddress string    `json:"installation_address" binding:"required"`
	Name                string    `json:"name" binding:"required"`
	CreatedAt           time.Time `json:"created_at"`
	UpdatedAt           time.Time `json:"updated_at"`
}

type CreateTerminalRequest struct {
	SerialNumber        string `json:"serial_number" binding:"required"`
	InstallationAddress string `json:"installation_address" binding:"required"`
	Name                string `json:"name" binding:"required"`
}

type UpdateTerminalRequest struct {
	SerialNumber        string `json:"serial_number,omitempty"`
	InstallationAddress string `json:"installation_address,omitempty"`
	Name                string `json:"name,omitempty"`
}

File: frontend/src/commonMain/kotlin/com/transportpay/data/models/Terminal.kt
package com.transportpay.data.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Terminal(
    val id: Int,
    @SerialName("serial_number") val serialNumber: String,
    @SerialName("installation_address") val installationAddress: String,
    val name: String?,
    @SerialName("created_at") val createdAt: String,
    @SerialName("updated_at") val updatedAt: String
)

@Serializable
data class CreateTerminalRequest(
    @SerialName("serial_number") val serialNumber: String,
    @SerialName("installation_address") val installationAddress: String,
    val name: String? = null
)

@Serializable
data class UpdateTerminalRequest(
    @SerialName("serial_number") val serialNumber: String? = null,
    @SerialName("installation_address") val installationAddress: String? = null,
    val name: String? = null
)
